src/ListActions/ListActions.less

```diff
@@ -1,115 +1,109 @@
+		margin-left: 0;
 
-	.moon-icon-button {
-		margin:0;
+		.enyo-locale-right-to-left & {
+			margin-left: @moon-spotlight-outset;
+			margin-right: 0;
+		}
 	}
-}
-.list-actions-drawer {
-	position: absolute;
-	top: 0;
-	left: 0;
-	bottom: 0;
-	right: 0;
-}
-.moon-header .list-actions-drawer {
-	top: -@moon-header-border-top-width;
-	bottom: -@moon-header-border-bottom-width;
-}
 
-/* Close button */
-.moon-icon-button.moon-list-actions-close {
-	position: absolute;
-	right: @moon-spotlight-outset;
-	top: @moon-spotlight-outset;
-	z-index: 2;
-}
-.enyo-locale-right-to-left .moon-icon-button.moon-list-actions-close {
-	right: auto;
-	left: @moon-spotlight-outset;
-}
+	.moon-hspacing > :last-child& .moon-icon,
+	.moon-hspacing > :last-child & .moon-icon {
+		margin-right: 0;
 
-/* Scroller */
-.moon-list-actions-scroller {
-	margin: 18px;
-	margin-right: @moon-icon-button-small-size + 18px;
-	padding: 0px;
-	z-index: 1;
-}
-.enyo-locale-right-to-left .moon-list-actions-scroller {
-	margin-right: 12px;
-	margin-left: @moon-icon-button-small-size + 18px;
+		.enyo-locale-right-to-left & {
+			margin-left: 0;
+			margin-right: @moon-spotlight-outset;
+		}
+	}
 }
 
-/* Action menu */
-.moon-list-actions-menu {
-	display: inline-block;
-	vertical-align: top;
-	width: 300px;		/* Do not change - used in JS */
-	min-width: 300px;	/* Do not change - used in JS */
-	float: right;
-	box-sizing: border-box;
-}
-.enyo-locale-right-to-left .moon-list-actions-menu {
-	float: left;
+.list-actions-left-hunits(@n) {
+	left: (@moon-icon-button-small-size * (@n - 1)) + (@moon-icon-button-small-size * @n) / 2 - @moon-icon-margin * @n / 2;
 }
 
-.moon-list-actions-drawer.proportional-width.stacked .moon-list-actions-menu {
-	width: 100% !important;
-}
-.moon-list-actions-drawer.stacked .moon-list-actions-menu {
-	margin-bottom: 18px;
-	clear: both;
-}
-.moon-list-actions-menu .enyo-scroller {
-	max-width: 100%;
+.list-actions-right-hunits(@n) {
+	right: (@moon-icon-button-small-size * (@n - 1)) + (@moon-icon-button-small-size * @n) / 2 - @moon-icon-margin * @n / 2;
 }
 
-/* Drawer */
-.moon-list-actions-drawer {
-	overflow: hidden;
-	z-index: 50;
-	pointer-events: none;
-	* {
-		pointer-events: auto;
+.moon-list-actions-popup {
+	& .moon-list-actions-popup-width {
+		width: 408px;
 	}
 
-	&.stacked .moon-list-actions-menu {
-		display: block;
+	& .moon-list-actions-scroller {
+		.enyo-scroller,
+		.enyo-data-list {
+			height: 300px;
+		}
 	}
-}
-.moon-list-actions-drawer-client {
-	position: absolute;
-	width: 100%;
-	height: 100%;
-	-webkit-transform: translateZ(0) translateY(100%);
-	transform: translateZ(0) translateY(100%);
 
-	&.open {
-		-webkit-transform: translateZ(0) translateY(0);
-		tranform: translateZ(0) translateY(0);
-	}
-	&.animated {
-		-webkit-transition: -webkit-transform .225s cubic-bezier(0.25, 0.1, 0.25, 1);
-		transition: transform .225s cubic-bezier(0.25, 0.1, 0.25, 1);
+	&.right {
+		&.list-actions-1h {
```

Comment:
Because we made the `list-actions-left-hunits` mixin very specific to the `ListActions` use case, what do you think about also incorporating the `before` and `after` pseudo element selectors as part of the mixin? And maybe we could consider passing the position property name as an additional parameter i.e. `.list-actions-h(1, left);` since the mixin is the same for both left and right, save for the property name. If we want to get really fancy, we could consider creating all of these selectors in a loop too: http://lesscss.org/features/#loops-feature.